src/main/scala/com/gu/Box.scala

```diff
@@ -12,7 +13,7 @@ abstract class Box[T] {
   def send(f: T => T): Unit
 
   def alter(t: T): Future[T]
-  def alter(f: T => T): Future[T]
+  def alter(f: T => Try[T])(implicit ec: ExecutionContext): Future[T]
```

Comment:
Should you add another methods rather than replacing `alter(T => T)`? Can it not be useful in some cases?

Reply:
meh 😐 I feel that is fulfilled by the `map` function, which has clear semantics.

Reply:
ok let's not dwell on that, I just did.